LayoutTest/TestCase/LYTLayoutFailingTestSnapshotRecorder.m

```diff
@@ -0,0 +1,233 @@
+#import "LYTConfig.h"
+#import "LYTLayoutTestCase.h"
+
+void SimpleLog(NSString *format, ...) {
+    // Log to the console without timestamp that NSLog gives us
+    va_list args;
+    va_start(args, format);
+    NSString *formattedString = [[NSString alloc] initWithFormat:format
+                                                       arguments:args];
+    va_end(args);
+    [[NSFileHandle fileHandleWithStandardOutput] writeData:[formattedString dataUsingEncoding:NSNEXTSTEPStringEncoding]];
+}
+
+@interface LYTLayoutFailingTestSnapshotRecorder ()
+
+@property (nonatomic) Class invocationClass;
+@property (nonatomic) NSMutableSet *failingTestsSnapshotFolders;
+
+@end
+
+@implementation LYTLayoutFailingTestSnapshotRecorder
+
++ (instancetype)sharedInstance {
+    static LYTLayoutFailingTestSnapshotRecorder *sharedInstance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[LYTLayoutFailingTestSnapshotRecorder alloc] init];
+    });
+    return sharedInstance;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.failingTestsSnapshotFolders = [NSMutableSet set];
+        XCTestObservationCenter *observationCenter = [XCTestObservationCenter sharedTestObservationCenter];
+        [observationCenter addTestObserver:self];
+    }
+    return self;
+}
+
+- (void)testBundleDidFinish:(NSBundle *)testBundle {
+    if (self.failingTestsSnapshotFolders.count > 0) {
+        SimpleLog(@"\nSnapshots of failing tests can be found in:\n");
+        for (NSString *path in self.failingTestsSnapshotFolders) {
+            SimpleLog(@"%@\n", path);
+        }
+        [self.failingTestsSnapshotFolders removeAllObjects];
+    }
+}
+
+- (void)testCase:(XCTestCase *)testCase didFailWithDescription:(NSString *)description inFile:(nullable NSString *)filePath atLine:(NSUInteger)lineNumber {
+    if ([testCase isKindOfClass:[LYTLayoutTestCase class]]) {
+        NSString *pathForSnapshots = [self commonRootPathForInvocationClass:[testCase class]];
+        pathForSnapshots = [pathForSnapshots stringByAppendingString:@"/index.html"];
+        [self.failingTestsSnapshotFolders addObject:pathForSnapshots];
+    }
+}
+
+- (void)startNewLogForClass:(Class)invocationClass {
+    self.invocationClass = invocationClass;
+    [self deleteCurrentFailingSnapshots];
+    [self createIndexHTMLFile];
+}
+
+- (void)saveImageOfView:(UIView *)view withData:(NSDictionary *)data fromInvocation:(NSInvocation *)invocation failureDescription:(NSString *)failureDescription {
+    NSString *imagePath = [self pathForImageWithWidth:view.frame.size.width height:view.frame.size.height data:data invocation:invocation];
+    if ([self shouldSaveImageOfViewatPath:imagePath withInvocation:invocation]) {
+        [self createDirectoryForInvocationIfNeeded:invocation];
+        UIImage *viewImage = [self renderLayer:view.layer];
+        [UIImagePNGRepresentation(viewImage) writeToFile:imagePath atomically:YES];
```

Comment:
Yeh it does return a BOOL. Could log the failure to the console but it will likely get lost in the test output. Other problem is that it'll still write to the index.html file but the image won't be there. In this case there is the option to only update the log if we successfully write to disk, otherwise update the log with a failure message. Happy to add that but like you say should almost never happen.